ui/summary.go

```diff
@@ -384,3 +385,44 @@ func (s *Summary) SummarizeMetrics(w io.Writer, indent string, data SummaryData)
 
 	s.summarizeMetrics(w, indent+"  ", data.Time, data.TimeUnit, data.Metrics)
 }
+
+func newJSONEncoder(w io.Writer) *json.Encoder {
+	encoder := json.NewEncoder(w)
+	encoder.SetIndent("", "    ")
+	return encoder
+}
+
+func summarizeGroupJSON(w io.Writer, group *lib.Group) error {
+	return newJSONEncoder(w).Encode(group)
+}
+
+func summarizeMetricsJSON(w io.Writer, t time.Duration, timeUnit string, metrics map[string]*stats.Metric) error {
+	data := make(map[string]interface{})
+	for name, m := range metrics {
+		m.Sink.Calc()
+		if sink, ok := m.Sink.(*stats.TrendSink); ok {
+			data[name] = sink.Format(t)
+			continue
+		}
+		data[name] = m.Sink.Format(t)
```

Comment:
Could this be shortened to:
```golang
data[name] = m.Sink.Format(t)
if _, ok := m.Sink.(*stats.TrendSink); ok {
	continue
}
```
?

Reply:
How can they be equivalent?

Reply:
Hhmm if you're referring to the type difference, you're dealing with `interface{}` so it shouldn't matter, no?

I'm not sure, I'd like to see tests to confirm, but it seems that the `Format(t)` call doesn't need to be repeated and can be called once.

Reply:
Ah ok

Reply:
Fixed